Gems/LyShine/Code/Source/UiCanvasManager.cpp

```diff
@@ -805,94 +808,98 @@ UiCanvasComponent* UiCanvasManager::FindEditorCanvasComponentByPathname(const AZ
-bool UiCanvasManager::HandleInputEventForInWorldCanvases([[maybe_unused]] const AzFramework::InputChannel::Snapshot& inputSnapshot, [[maybe_unused]] const AZ::Vector2& viewportPos)
+bool UiCanvasManager::HandleInputEventForInWorldCanvases(const AzFramework::InputChannel::Snapshot& inputSnapshot, const AZ::Vector2& viewportPos)
 {
-    // First we need to construct a ray from the either the center of the screen or the mouse position.
-    // This requires knowledge of the camera
-    // for initial testing we will just use a ray in the center of the viewport
+    if (!AZ::Interface<ILyShine>::Get())
+    {
+        return false;
+    }
+
+    // First we need to construct a ray from either the center of the screen or the mouse position
+    CLyShine* lyShine = static_cast<CLyShine*>(AZ::Interface<ILyShine>::Get());
+    auto viewportContext = lyShine->GetUiRenderer()->GetViewportContext();
+    AZ::Transform cameraTransform = viewportContext->GetCameraTransform();
+
+    const float rayLength = 500.0f;
+    AZ::Vector3 rayOrigin(0.0f);
+    AZ::Vector3 endpoint(0.0f, rayLength, 0.0f);
+    rayOrigin = cameraTransform.TransformPoint(rayOrigin);
+
+    // If the mouse cursor is visible we will assume that the ray should be in the direction of the
+    // mouse pointer. This is a temporary solution. A better solution is to be able to configure the
+    // LyShine system to say how ray input should be handled.
+    bool isCursorVisible = false;
+    UiCursorBus::BroadcastResult(isCursorVisible, &UiCursorInterface::IsUiCursorVisible);
+    if (!isCursorVisible)
+    {
+        endpoint = cameraTransform.TransformPoint(endpoint);
+    }
+    else
+    {
+        AzFramework::WindowSize viewportSize = viewportContext->GetViewportSize();
+        AzFramework::ScreenSize screenSize(viewportSize.m_width, viewportSize.m_height);
+        AzFramework::ScreenPoint screenPosition(static_cast<int>(viewportPos.GetX()), static_cast<int>(viewportPos.GetY()));
+
+        AzFramework::CameraState cameraState = AzFramework::CreateDefaultCamera(cameraTransform, screenSize);
+        const AZ::Matrix4x4& viewToClipMatrix = viewportContext->GetCameraProjectionMatrix();
+        AzFramework::SetCameraClippingVolumeFromPerspectiveFovMatrixRH(cameraState, viewToClipMatrix);
+
+        endpoint = AzFramework::ScreenToWorld(
+            screenPosition,
+            cameraState
+        );
+
+        AZ::Vector3 rayDirection = (endpoint - rayOrigin).GetNormalized();
+        rayDirection.SetLength(rayLength);
+        endpoint = rayOrigin + rayDirection;
+    }
+
+    // Check if the ray intersects any entities
+    AzFramework::EntityContextId gameContextId;
+    AzFramework::GameEntityContextRequestBus::BroadcastResult(gameContextId,
+        &AzFramework::GameEntityContextRequests::GetGameEntityContextId);
+        
+    AzFramework::RenderGeometry::RayRequest request;
+    request.m_startWorldPosition = rayOrigin;
+    request.m_endWorldPosition = endpoint;
```

Comment:
If another entity is in front of the entity with the UI canvas on it, then we want to block input for that UI canvas, so that's why there's currently no filter.